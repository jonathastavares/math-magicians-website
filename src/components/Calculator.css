.calculator {
  display: grid;
}

button {
  border-style: solid;
  border-width: 1px;
  cursor: pointer;
  padding: 10px;
  border-color: #bdbdbd;
}

.quote {
  background-color: gray;
  color: white;
  height: 50px;
  text-align: right;
  border-style: unset;
  font-size: 30px;
}

.quote::placeholder {
  color: white;
}

.number {
  background-color: #e0e0e0;
}

.number:hover {
  background-color: #b8b8b8;
}

.number-section {
  display: grid;
  grid-template-columns: auto auto auto;
}

.operation-section {
  display: flex;
  flex-direction: column;
}

.operation {
  background-color: orange;
}

.operation:hover {
  background-color: rgb(211, 137, 0);
}

.buttons {
  display: grid;
  grid-template-columns: 75% auto;
}

.orange {
  color: orange;
  text-decoration-color: black !important;
}
