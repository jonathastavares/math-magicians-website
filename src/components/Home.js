import './Home.css';

function Home() {
  return (
    <div className="text-center">
      <h1 className="fw-bold fs-1 orange mt-5 text-decoration-underline">Welcome to our page</h1>
      <p className="mt-5 fs-5">
        In this website, you will be able to use our incredible Calculator!
      </p>
    </div>
  );
}

export default Home;
