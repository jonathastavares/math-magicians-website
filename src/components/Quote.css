.orange {
  color: orange;
  text-decoration-color: black !important;
}
