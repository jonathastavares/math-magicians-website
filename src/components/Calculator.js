import React, { useState } from 'react';
import PropTypes from 'prop-types';
import calculate from '../logic/calculate';
import './Calculator.css';

function Op({ operation, cb }) {
  return (
    <button type="button" className="operation" onClick={cb}>
      {operation}
    </button>
  );
}

function Number({ num, cb }) {
  const style = {};
  if (num === '0') {
    style['grid-column'] = 'span 2';
  }

  return (
    <button type="button" style={style} className="number" onClick={cb}>
      {num}
    </button>
  );
}

Number.propTypes = {
  num: PropTypes.string.isRequired,
  cb: PropTypes.func.isRequired,
};

Op.propTypes = {
  operation: PropTypes.string.isRequired,
  cb: PropTypes.func.isRequired,
};

function Calculator() {
  const [state, setState] = useState({
    total: 0,
    next: 0,
    operation: '',
  });
  const numberSectionButtons = [];
  ['AC', '+/-', '%', 7, 8, 9, 4, 5, 6, 1, 2, 3, 0, '.'].forEach((el) => {
    numberSectionButtons.push(<Number
      num={`${el}`}
      cb={() => {
        setState((old) => calculate(old, `${el}`));
      }}
    />);
  });

  return (
    <div className="d-flex flex-column align-items-center w-100">
      <h1 className="fw-bold fs-1 orange mt-5 text-decoration-underline">Lets do some math!!</h1>
      <div className="calculator mt-5 border border-5 border-light shadow-lg">
        <input className="quote" placeholder="0" value={state.next ? state.next : state.total} readOnly />
        <div className="buttons">
          <div className="number-section">
            {numberSectionButtons}
          </div>
          <div className="operation-section">
            <Op operation="รท" cb={() => setState((old) => ({ ...old, ...calculate(old, 'รท') }))} />
            <Op operation="x" cb={() => setState((old) => ({ ...old, ...calculate(old, 'x') }))} />
            <Op operation="-" cb={() => setState((old) => ({ ...old, ...calculate(old, '-') }))} />
            <Op operation="+" cb={() => setState((old) => ({ ...old, ...calculate(old, '+') }))} />
            <Op operation="=" cb={() => setState((old) => ({ ...old, ...calculate(old, '=') }))} />
          </div>
        </div>
      </div>
    </div>
  );
}

export default Calculator;
