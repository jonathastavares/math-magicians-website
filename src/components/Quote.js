import './Quote.css';

function Quote() {
  return (
    <div className="fs-3 text-center text-decoration-underline fw-bold d-flex justify-content-center mt-5">
      <p className="orange">
        Mathematics is not about numbers, equations, computations, or algorithms.
        It is about understanding.
        - William Paul Thurston
      </p>
    </div>
  );
}

export default Quote;
