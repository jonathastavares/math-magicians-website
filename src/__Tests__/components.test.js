import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import Quote from '../components/Quote';
import Home from '../components/Home';
import Calculator from '../components/Calculator';

describe('Testing Components', () => {
  it('Tests Home.js', () => {
    const tree = render(<Home />);
    expect(tree).toMatchSnapshot();
  });

  it('Tests Quote.js', () => {
    const tree = render(<Quote />);
    expect(tree).toMatchSnapshot();
  });

  it('Tests Calculator.js', () => {
    const tree = render(<Calculator />);
    expect(tree).toMatchSnapshot();
    fireEvent.click(screen.getByText('1'));
    expect(screen.getByTestId('result')).toHaveValue('1');
    fireEvent.click(screen.getByText('+'));
    fireEvent.click(screen.getByText('2'));
    fireEvent.click(screen.getByText('='));
    expect(screen.getByTestId('result')).toHaveValue('3');
  });
});
