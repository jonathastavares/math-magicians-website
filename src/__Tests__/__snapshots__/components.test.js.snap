// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Components Tests Calculator.js 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex flex-column align-items-center w-100"
      >
        <h1
          class="fw-bold fs-1 orange mt-5 text-decoration-underline"
        >
          Lets do some math!!
        </h1>
        <div
          class="calculator mt-5 border border-5 border-light shadow-lg"
        >
          <input
            class="quote"
            data-testid="result"
            placeholder="0"
            readonly=""
            value="0"
          />
          <div
            class="buttons"
          >
            <div
              class="number-section"
            >
              <button
                class="number"
                type="button"
              >
                AC
              </button>
              <button
                class="number"
                type="button"
              >
                +/-
              </button>
              <button
                class="number"
                type="button"
              >
                %
              </button>
              <button
                class="number"
                type="button"
              >
                7
              </button>
              <button
                class="number"
                type="button"
              >
                8
              </button>
              <button
                class="number"
                type="button"
              >
                9
              </button>
              <button
                class="number"
                type="button"
              >
                4
              </button>
              <button
                class="number"
                type="button"
              >
                5
              </button>
              <button
                class="number"
                type="button"
              >
                6
              </button>
              <button
                class="number"
                type="button"
              >
                1
              </button>
              <button
                class="number"
                type="button"
              >
                2
              </button>
              <button
                class="number"
                type="button"
              >
                3
              </button>
              <button
                class="number"
                type="button"
              >
                0
              </button>
              <button
                class="number"
                type="button"
              >
                .
              </button>
            </div>
            <div
              class="operation-section"
            >
              <button
                class="operation"
                type="button"
              >
                รท
              </button>
              <button
                class="operation"
                type="button"
              >
                x
              </button>
              <button
                class="operation"
                type="button"
              >
                -
              </button>
              <button
                class="operation"
                type="button"
              >
                +
              </button>
              <button
                class="operation"
                type="button"
              >
                =
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex flex-column align-items-center w-100"
    >
      <h1
        class="fw-bold fs-1 orange mt-5 text-decoration-underline"
      >
        Lets do some math!!
      </h1>
      <div
        class="calculator mt-5 border border-5 border-light shadow-lg"
      >
        <input
          class="quote"
          data-testid="result"
          placeholder="0"
          readonly=""
          value="0"
        />
        <div
          class="buttons"
        >
          <div
            class="number-section"
          >
            <button
              class="number"
              type="button"
            >
              AC
            </button>
            <button
              class="number"
              type="button"
            >
              +/-
            </button>
            <button
              class="number"
              type="button"
            >
              %
            </button>
            <button
              class="number"
              type="button"
            >
              7
            </button>
            <button
              class="number"
              type="button"
            >
              8
            </button>
            <button
              class="number"
              type="button"
            >
              9
            </button>
            <button
              class="number"
              type="button"
            >
              4
            </button>
            <button
              class="number"
              type="button"
            >
              5
            </button>
            <button
              class="number"
              type="button"
            >
              6
            </button>
            <button
              class="number"
              type="button"
            >
              1
            </button>
            <button
              class="number"
              type="button"
            >
              2
            </button>
            <button
              class="number"
              type="button"
            >
              3
            </button>
            <button
              class="number"
              type="button"
            >
              0
            </button>
            <button
              class="number"
              type="button"
            >
              .
            </button>
          </div>
          <div
            class="operation-section"
          >
            <button
              class="operation"
              type="button"
            >
              รท
            </button>
            <button
              class="operation"
              type="button"
            >
              x
            </button>
            <button
              class="operation"
              type="button"
            >
              -
            </button>
            <button
              class="operation"
              type="button"
            >
              +
            </button>
            <button
              class="operation"
              type="button"
            >
              =
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing Components Tests Home.js 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="text-center"
      >
        <h1
          class="fw-bold fs-1 orange mt-5 text-decoration-underline"
        >
          Welcome to our page
        </h1>
        <p
          class="mt-5 fs-5"
        >
          In this website, you will be able to use our incredible Calculator!
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="text-center"
    >
      <h1
        class="fw-bold fs-1 orange mt-5 text-decoration-underline"
      >
        Welcome to our page
      </h1>
      <p
        class="mt-5 fs-5"
      >
        In this website, you will be able to use our incredible Calculator!
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing Components Tests Quote.js 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="fs-3 text-center text-decoration-underline fw-bold d-flex justify-content-center mt-5"
      >
        <p
          class="orange"
        >
          Mathematics is not about numbers, equations, computations, or algorithms. It is about understanding. - William Paul Thurston
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="fs-3 text-center text-decoration-underline fw-bold d-flex justify-content-center mt-5"
    >
      <p
        class="orange"
      >
        Mathematics is not about numbers, equations, computations, or algorithms. It is about understanding. - William Paul Thurston
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
